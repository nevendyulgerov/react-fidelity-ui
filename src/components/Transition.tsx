import * as React from 'react';
import * as PropTypes from 'prop-types';
import classNames from 'classnames';
import { isNonEmptyStr } from '../utils';

export type AnimationProps = {
  children: React.ReactNode,
  type?: string | null,
  className?: string | null,
  [key: string]: any
};

const Transition = (props: AnimationProps) => {
  const { children, type, className, ...restProps } = props;
  const componentClassName: string = classNames({
    transition: true,
    // @ts-ignore
    [`transition--${type}`]: isNonEmptyStr(type),
    // @ts-ignore
    [className]: isNonEmptyStr(className)
  });

  return (
    <div
      {...restProps}
      className={componentClassName}
    >
      {children}
    </div>
  );
};

Transition.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
    PropTypes.node,
    PropTypes.arrayOf(PropTypes.node)
  ]).isRequired,
  type: PropTypes.oneOf([
    'fade',
    'fade-in-down',
    'fade-in-left',
    'fade-in-right',
    'fade-in-up',
    'fade-out',
    'slide-in-down',
    'slide-in-left',
    'slide-in-right',
    'slide-in-up',
    'slide-out-down',
    'slide-out-left',
    'slide-out-right',
    'slide-out-up',
    'spin'
  ]),
  className: PropTypes.string
};

Transition.defaultProps = {
  type: 'fade',
  className: null
};

export default Transition;
