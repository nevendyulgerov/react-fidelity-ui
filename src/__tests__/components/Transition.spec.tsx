import * as React from 'react';
import { render } from '@testing-library/react';
import Transition from '../../components/Transition';

describe('Components / Transition', () => {
  it('matches snapshot', () => {
    const { baseElement } = render(<Transition><span>Transition</span></Transition>);
    expect(baseElement).toMatchSnapshot();
  });
});
