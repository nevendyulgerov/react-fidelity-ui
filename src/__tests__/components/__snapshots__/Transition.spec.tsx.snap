// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components / Transition matches snapshot 1`] = `
<body>
  <div>
    <div
      class="transition transition--fade"
    >
      <span>
        Transition
      </span>
    </div>
  </div>
</body>
`;
